### PR TITLE
3.3_higher-order_functions.ipynb: small fixes

### PR DESCRIPTION
**Pull request by lmy441900**
These are some personal fixes during my going through the Chisel Bootcamp. :smile:

### CODE REVIEW

File: 3.3_higher-order_functions.ipynb [hunk 1]
```diff
@@ -87,7 +87,7 @@
     "\n",
     "---\n",
     "# Functions as Arguments\n",
-    "Formally, functions like `map` and `reduce` are called _higher-order functions_: they are functions that take functions as arguments.\n",
+    "Formally, functions like `map` and `reduce` are called _higher-order functions_ : they are functions that take functions as arguments.\n",
```
Comment (edwardcwang) [inline 3.3_higher-order_functions.ipynb]: This change isn't quite necessary, no?

**Comment by edwardcwang** [comment]
Thanks a lot for the fixes!

File: 3.3_higher-order_functions.ipynb (hunk 1 above)
Comment (lmy441900) [reply 3.3_higher-order_functions.ipynb]: If you think it's not necessary then I can remove this commit, however it does cause a rendering problem on JupyterHub (italic fonts are not rendered and underscores are shown). This is a small hack.

Comment (edwardcwang) [reply 3.3_higher-order_functions.ipynb]: Good catch! They must have a bug somewhere...

**Pull request merged**